exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcConnectionImpl.kt

```diff
@@ -15,8 +15,11 @@ import java.sql.PreparedStatement
 class JdbcConnectionImpl(override val connection: Connection) : ExposedConnection<Connection> {
 
     // Oracle driver could throw excpection on catalog
-    override val catalog: String by lazy {
-        try { connection.catalog } catch (_: Exception) { null } ?: connection.metaData.userName ?: ""
+    override val catalog: String
+    get() =  try { connection.catalog } catch (_: Exception) { null } ?: connection.metaData.userName ?: ""
+
+    override val schema: String by lazy {
+        connection.schema
     }
```

Comment:
Afaik, some drivers (e.g. MySQL) could throw `AbstractMethodException` on an attempt to get `schema` from a connection. Also, I think it should be getter to as you add the possibility to change a schema.